### PR TITLE
Katetc/autogenio rebased

### PR DESCRIPTION
**Pull request by Katetc**
Changes to autogenerate io files each time cism is built in cesm. Also tested and updated to include python 3 compatibility. 

This code has not been fully tested with an aux suite as Cheyenne's pbs system is currently down. However, multiple tests of the build system on Cheyenne and Hobart have passed, and the generate files compared successfully. So, I think it's ok.

### CODE REVIEW

File: utils/build/generate_ncvars.py [hunk 1]
```diff
@@ -23,7 +23,12 @@
 
 # python script used to generate source code files given a variable definition file
 
-import ConfigParser, sys, time, string,re, os.path
+import sys, time, string,re, os.path
+import argparse
```
Comment (billsacks) [inline utils/build/generate_ncvars.py]: argparse is unused; you should remove this import statement

File: CMakeLists.txt [hunk 2]
```diff
@@ -179,37 +178,6 @@ INCLUDE_DIRECTORIES(${CMAKE_Fortran_MODULE_DIRECTORY})
...
-                    WORKING_DIRECTORY ${CISM_BINARY_DIR}
-                    OUTPUT_FILE ${CISM_BINARY_DIR}/autogen-for-glint-and-glad.log)
- 
-    MESSAGE(">> see ${CISM_BINARY_DIR}/autogen-for-glint-and-glad.log")
-  ENDIF()
-
-ELSE()
-  # Simply copy the default io files into the build directory
-  MESSAGE(">> Calling utils/build/autocopy-io-to-build-dir")
-  EXECUTE_PROCESS(COMMAND  ${CISM_SOURCE_DIR}/utils/build/autocopy-io-to-build-dir
```
[the remark this answers is not in the document]
Comment (Katetc) [reply CMakeLists.txt]: Interestingly, the auto-copy script is still used (L183 of CMakeLists.txt). In the log it says that it copies over glimmer_vers.F90, which is a file that just holds the glimmer version information. I think we discussed if this should be required or not and didn't come to any conclusions. It is required to build CISM, even in CESM, for some reason. I can add an issue that we need to revisit this file, but I don't think that removing this script and file are scoped in this PR/project (as in, I want to get this in and I don't think we should worry about glimmer_vers.F90 right now).

[the remark this answers is not in the document]
Comment (Katetc) [reply CMakeLists.txt]: Wait, read the name of the file wrong. Ooops. autocopy-io-to-build-dir and autocopy-to-build-dir are two different files. You want me to remove the first one. No problemo.

File: CMakeLists.txt [hunk 3]
```diff
@@ -336,13 +311,38 @@ ELSE(${CISM_SERIAL_MODE})
...
+		COMMAND ${CISM_SOURCE_DIR}/utils/build/autogenerate-in-build-dir
+		${CISM_SOURCE_DIR} 
+		DEPENDS ${CISM_SOURCE_DIR}/libglide/glide_vars.def 
+		        ${CISM_SOURCE_DIR}/libglimmer/ncdf_template.F90.in
+)
+
+# Auto-generate the *_io.F90 files in the build directory:
+MESSAGE(">> Calling utils/build/autogen-for-glint-and-glad-in-build-dir")
+
+ADD_CUSTOM_COMMAND(
```
[the remark this answers is not in the document]
Comment (Katetc) [reply CMakeLists.txt]: Hmm, this one I don't really understand. I thought the whole point of this was to treat the CESM io files the same way we treat them in CISM standalone. So, yes, regardless of which "front end" is being used, we want to generate the io files, I think? I gave a quick test where I slapped that IF statement back around the autogen code for glint and glad, and, no, CISM won't build without the _io.F90 files present. So, maintaining that conditional logic is not trivial. And I don't really understand why you would want to, either.

[the remark this answers is not in the document]
Comment (billsacks) [reply CMakeLists.txt]: Okay, it looks like the glint and glad files are always added to the build, so it looks like what you have is correct.

**Pull request merged**